pkg/cloud/azure/azure_kms.go

```diff
@@ -46,6 +48,14 @@ func init() {
 	cloud.RegisterKMSFromURIFactory(MakeAzureKMS, kmsScheme)
 }
 
+type AzureAuth int32
+
+const (
```

Comment:
instead of declaring a new type here, why not use the enum in the external_storage.proto and use it in the KMSURIParams struct? You would need to move the enum declaration in the proto outside the Azure `message`.

Reply:
Raised in Slack, but I have two reasons:
1. They're not _exactly_ the same enum; KMS does not in fact support the same auth options as Blob Storage, for instance excluding account key.
2. Depending on a storage file here feels like a mistaken dependency.